syncd/syncd_saiswitch.cpp

```diff
@@ -936,6 +936,12 @@ void SaiSwitch::saiDiscover(
             {
                 sai_object_id_t oid = attr.value.objlist.list[i];
 
+                if(SAI_NULL_OBJECT_ID == oid)
+                {
+                    SWSS_LOG_DEBUG("oid is NULL for %s index %u", md->attridname, i);
+                    continue;
+                }
+
```

Comment:
there is no dereference prevention here, if you look at line 939:
 ot = sai_object_type_query(oid); this should return SAI_OBJECT_TYPE_NULL if oid is null,
and SAI_NULL_OBJECT_ID  is not expected value on any attr.value.objlist, so at this point throw should be made ad line 943, indicating that this is vendor bug
and this entire check you added is not necessary

Reply:
Thanks for the comment. Will decline this PR.